contracts/strategies/allocation/QFOffchainCalc.sol

```diff
@@ -0,0 +1,194 @@
+
+        // Add the application to the applications mapping
+        applications[sender] = application;
+
+        // Add the application to the applicationIdToApplication mapping
+        applicationIdToApplication[application.applicationId] = application;
+
+        emit ApplicationCreated(sender, application.applicationId);
+
+        return application.applicationId;
+    }
+
+    // Todo: Why don't we have a adddress sender here?
+    /// @notice Returns the status of the application
+    /// @param applicationId The applicationId of the application
+    function getApplicationStatus(uint256 applicationId) external view override returns (ApplicationStatus) {
+        return globalStatusMap[msg.sender][applicationIdToApplication[applicationId].status];
+    }
+
+    /// @notice Allocates votes to the application(s)
+    /// @param _data The data to be decoded
+    /// @param sender The sender of the transaction
+    function allocate(bytes memory _data, address sender) external payable override onlyOwner returns (uint256) {
+        if (votingStartTime > block.timestamp || votingEndTime < block.timestamp) {
+            revert VOTING_NOT_OPEN();
+        }
+
+        // Decode the _data
+        (uint256 vote, address token) = abi.decode(_data, (uint256, address));
+
+        // Todo: Do we pull the poolId from the _data?
+        // allo.allocate(_poolId, _data);
+
+        Application storage application = applications[sender];
+        application.votes += vote;
+
+        emit ApplicationVoted(sender, applications[sender].applicationId, vote, token, application.recipient);
+
+        // Note: Return the total votes for Application?
+        return application.votes;
+    }
+
+    /// @notice Generates the payouts for the application(s)
+    function generatePayouts() external payable override onlyOwner returns (bytes memory) {}
+
+    /// ====================================
+    /// ========= Custom Functions =========
+    /// ====================================
+
+    /// @notice Creates a pool
+    /// @param _identityId The identityId of the pool
+    /// @param _distributionStrategy The distribution strategy of the pool
+    /// @param _token The token of the pool
+    /// @param _amount The amount of the pool
+    /// @param _metadata The metadata of the pool
+    function createPool(
+        bytes32 _identityId,
+        address _distributionStrategy,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata
+    ) external onlyOwner {
+        allo.createPool(_identityId, address(this), payable(_distributionStrategy), _token, _amount, _metadata);
+    }
+
+    /// @notice Review applications
+    /// @dev Pool admin can only set applications to `ACCEPTED`, `REJECTED` or `REAPPLIED`
+    function reviewApplications() external onlyOwner {
+        // Todo: Pool admin can only set applications to `ACCEPTED`, `REJECTED` or `REAPPLIED`
```

Comment:
The pool admin shouldn't be able to set the "reapplied" status — that should happen automatically when a project reapplies